pkg/scale/encode.go

```diff
@@ -374,9 +516,12 @@ func (es *encodeState) encodeUint(i uint) (err error) {
 // encodeUint128 encodes a Uint128
 func (es *encodeState) encodeUint128(i *Uint128) (err error) {
 	if i == nil {
-		err = fmt.Errorf("uint128 is nil: %v", i)
+		err = fmt.Errorf("%w: nil", ErrEncodeUint128)
 		return
 	}
 	err = binary.Write(es, binary.LittleEndian, padBytes(i.Bytes(), binary.LittleEndian))
+	if err != nil {
+		err = fmt.Errorf("encoding Uint128: %w", err)
```

Comment:
(assuming `Uint128` have a string representation `String() string`?)

```suggestion
		err = fmt.Errorf("writing binary for Uint128 %s: %w", i, err)
```